x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/action/bulk/TransportBulkShardOperationsAction.java

```diff
@@ -151,20 +183,36 @@ protected BulkShardOperationsResponse newResponseInstance() {
     /**
      * Custom write result to include global checkpoint after ops have been replicated.
      */
-    static class CcrWritePrimaryResult extends WritePrimaryResult<BulkShardOperationsRequest, BulkShardOperationsResponse> {
+    static final class CcrWritePrimaryResult extends WritePrimaryResult<BulkShardOperationsRequest, BulkShardOperationsResponse> {
+        final long waitingForGlobalCheckpoint;
 
-        CcrWritePrimaryResult(BulkShardOperationsRequest request, Translog.Location location, IndexShard primary, Logger logger) {
+        CcrWritePrimaryResult(BulkShardOperationsRequest request, Translog.Location location, IndexShard primary,
+                              long waitingForGlobalCheckpoint, Logger logger) {
             super(request, new BulkShardOperationsResponse(), location, null, primary, logger);
+            this.waitingForGlobalCheckpoint = waitingForGlobalCheckpoint;
         }
 
         @Override
         public synchronized void respond(ActionListener<BulkShardOperationsResponse> listener) {
-            final BulkShardOperationsResponse response = finalResponseIfSuccessful;
-            final SeqNoStats seqNoStats = primary.seqNoStats();
-            // return a fresh global checkpoint after the operations have been replicated for the shard follow task
```

Comment:
why lose the comment?

Reply:
I pushed https://github.com/elastic/elasticsearch/pull/34288/commits/146c8bb989e7f5dc37377ef33bf5729f1b68813a.